packages/checkout/widgets-lib/src/widgets/connect/views/SwitchNetworkZkEVM.tsx

```diff
@@ -38,7 +39,63 @@ export function SwitchNetworkZkEVM() {
       controlType: 'Button',
     });
 
+    if (!provider.provider.request) return;
+
+    const currentChainId = provider.provider.request({ method: 'eth_chainId', params: [] });
+    // eslint-disable-next-line radix
+    const parsedChainId = parseInt(currentChainId.toString());
+
+    if (parsedChainId === getL2ChainId(checkout.config)) {
+      connectDispatch({
+        payload: {
+          type: ConnectActions.SET_PROVIDER,
+          provider,
+        },
+      });
+
+      viewDispatch({
+        payload: {
+          type: ViewActions.UPDATE_VIEW,
+          view: {
+            type: ConnectWidgetViews.SUCCESS,
+          },
+        },
+      });
+
+      return;
+    }
+
+    const walletName = (provider.provider as any)?.session?.peer?.metadata?.name.toLowerCase();
```

Comment:
Does it make sense to execute this if isWalletConnectProvider(provider) is false?